extensions/amp-story/1.0/amp-story.js

```diff
@@ -2661,10 +2662,17 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   initializeStoryNavigationPath_() {
-    this.storeService_.dispatch(
-      Action.SET_NAVIGATION_PATH,
-      getHistoryState(this.win, HistoryState.NAVIGATION_PATH) || []
+    let navigationPath = getHistoryState(
+      this.win,
+      HistoryState.NAVIGATION_PATH
     );
+    if (
+      !navigationPath ||
+      !navigationPath.every((pageId) => this.isActualPage_(pageId))
```

Comment:
Nit: you can exit the loop on the first non actual page

```suggestion
      navigationPath.some((pageId) => !this.isActualPage_(pageId))
```

Reply:
https://jsfiddle.net/jqy431mh/1 `.every` also exits the loop on the first non-actual page

Reply:
Makes sense 👍